iotit.go

```diff
@@ -41,46 +54,140 @@ func init() {
 
 	f, err := os.OpenFile(fmt.Sprintf("/tmp/%s.log", progName), os.O_APPEND|os.O_CREATE|os.O_RDWR, 0666)
 	if err != nil {
-		logrus.Error("error opening file: %v", err)
+		logrus.Errorf("error opening file: %v", err)
 		return
 	}
 
 	logrus.SetOutput(f)
+
+	initCommands()
 }
 
 func main() {
 	deviceType := flag.String("dev", "", "")
-	h1 := flag.Bool("help", false, "")
-	h2 := flag.Bool("h", false, "")
-
-	v1 := flag.Bool("version", false, helpInfo)
-	v2 := flag.Bool("v", false, "")
-
-	u := flag.String("update", "", "")
 
 	flag.Parse()
 
-	if *h1 || *h2 {
-		fmt.Println(helpInfo)
+	if commandsHandler(flag.Args()) {
 		return
 	}
+	device.Init(*deviceType)
+}
+
+func commandsHandler(args []string) bool {
+	for _, arg := range args {
+		if c, ok := commands[arg]; ok {
+			c()
+			return true
+		}
+	}
 
-	if *v1 || *v2 {
+	return false
+}
+
+func initCommands() {
+	v := func() {
 		if Version == "" {
 			Version = "no version"
 		}
+
 		fmt.Println(progName, Version)
-		return
 	}
 
-	if *u != "" {
+	h := func() {
+		fmt.Println(helpInfo)
+	}
+
+	i := func() {
+		logrus.Debug("Checking ", installPath, progName)
+		if _, err := os.Stat(installPath + progName); os.IsNotExist(err) {
+			p, err := os.Executable()
```

Comment:
__[gometalinter]__
* [interfacer] Executable not declared by package os
* [unconvert] Executable not declared by package os

<!-- Comment by [SideCI](https://sideci.com) -->